bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/Parameters.scala

```diff
@@ -108,84 +108,79 @@ sealed trait CalculationMethod {
+                             exposureTime: Double,
+                             sourceFraction: Double,
+                             offset: Double) extends Imaging with S2NMethod {
   def isSpectroscopy: Boolean = false
   def isImaging: Boolean = true
 }
 
-// Return the number of exposures (Integration Time) given the exposure time, desired S/N, etc.
-final case class ImagingInt(
-                    sigma: Double,
-                    exposureTime: Double,
-                    coadds: Option[Int],
-                    sourceFraction: Double,
-                    offset: Double) extends Imaging with IntMethod {
-
-  def isSpectroscopy: Boolean = false
-  def isImaging: Boolean = true
+// Calculations that return exposure time to achivee a given SN
+sealed trait IntMethod extends CalculationMethod {
+  def sigma: Double
 }
 
 // Return the exposure time and number of exposures given the desired S/N
-final case class ImagingExp(
-                    sigma: Double,
-                    coadds: Option[Int],
-                    sourceFraction: Double,
-                    offset: Double) extends Imaging with IntMethod {
+final case class ImagingInt(
+                             sigma: Double,
+                             coadds: Option[Int],
+                             sourceFraction: Double,
+                             offset: Double) extends Imaging with IntMethod {
   val exposureTime: Double = 60
   def isSpectroscopy: Boolean = false
   def isImaging: Boolean = true
 }
 
-final case class SpectroscopyS2N(
-                                  exposures: Int,
-                                  coadds: Option[Int],
-                                  exposureTime: Double,
-                                  sourceFraction: Double,
-                                  offset: Double,
-                                  wavelengthAt: Option[Double] // wavelength in nanometers
-                  ) extends Spectroscopy with S2NMethod {
-  val atWithDefault: Double = wavelengthAt.getOrElse(0) // Not a very nice default, but we'd be outside any CCD and just return 0 sn
-  def isSpectroscopy: Boolean = true
-  def isImaging: Boolean = false
-}
-
 // Return the spectroscopic integration time (exposure time & number of exposures) given the desired S/N
 final case class SpectroscopyInt(
                                   sigma: Double,
                                   wavelengthAt: Double,
                                   coadds: Option[Int],
                                   sourceFraction: Double,
-                                  offset: Double) extends Spectroscopy with SpecIntMethod with S2NMethod {
+                                  offset: Double) extends Spectroscopy with IntMethod {
   val exposureTime: Double = 1200
   val exposures: Int = 1
   def isSpectroscopy: Boolean = true
   def isImaging: Boolean = false
 }
 
 
+// Return the number of exposures (count) given the exposure time, desired S/N, etc.
+final case class ImagingExpCount(
```

Comment:
This is another source of confusion as this is neither `IntMethod` nor `S2NMethod`